addons/project/models/project.py

```diff
@@ -0,0 +1,633 @@
+    alias_model = fields.Selection(_alias_models, "Alias Model", select=True, required=True, default='project.task',
+                                    help="The kind of document created when an email is received on this project's email alias")
+    privacy_visibility = fields.Selection(_visibility_selection, 'Privacy / Visibility', required=True, default='employees',
+        help="Holds visibility of the tasks or issues that belong to the current project:\n"
+                "- Portal : employees see everything;\n"
+                "   if portal is activated, portal users see the tasks or issues followed by\n"
+                "   them or by someone of their company\n"
+                "- Employees Only: employees see all tasks or issues\n"
+                "- Followers Only: employees see only the followed tasks or issues; if portal\n"
+                "   is activated, portal users see the followed tasks or issues.")
+    state = fields.Selection([('draft','New'),
+                              ('open','In Progress'),
+                              ('cancelled', 'Cancelled'),
+                              ('pending','Pending'),
+                              ('close','Closed')],
+                              'Status', required=True, copy=False, default='open')
+    doc_count = fields.Integer(compute='_compute_get_attached_docs', string="Number of documents attached")
+    date_start = fields.Date('Start Date')
+    date = fields.Date('Expiration Date', select=True, track_visibility='onchange')
+
+    _defaults = {
+        'use_tasks': True
+    }
+
+    @api.multi
+    def _compute_task_count(self):
+        task_data = self.env['project.task'].read_group([('project_id', 'in', self.ids),'|', ('stage_id.fold', '=', False), ('stage_id', '=', None)], ['project_id'], ['project_id'])
+        mapped_data = {task['project_id'][0]: task['project_id_count'] for task in task_data}
+        for project in self:
+            project.task_count = mapped_data.get(project.id)
+
+    @api.multi
+    def _compute_get_attached_docs(self):
+        Attachment = self.env['ir.attachment']
+        attachment_data = Attachment.read_group([('res_model', '=', 'project.project'), ('res_id', 'in', self.ids)], ['res_id'], ['res_id'])
+        project_attachments = {attachment['res_id']: attachment['res_id_count'] for attachment in attachment_data}
+        Task = self.env['project.task']
+        for project in self:
+            tasks = Task.search([('project_id', '=', project.id)])
+            task_attachments = Attachment.search([('res_model', '=', 'project.task'), ('res_id', 'in', tasks.ids)], count=True)
+            project.doc_count = project_attachments.get(project.id, 0) + (task_attachments and task_attachments.id or 0)
+
+    @api.constrains('date_start', 'date')
+    def _check_dates(self):
+        if self.date_start and self.date and self.date_start > self.date:
```

Comment:
is it work self.date_start when self contain multiple recordset ?  @api.constraints without decorator take default @api.multi . so iteration required.